src/SdkConfig.ts

```diff
@@ -78,11 +77,11 @@ export default class SdkConfig {
         return SdkConfig.fallback.get(key, altCaseName);
     }
 
-    public static getObject<K extends KeysWithObjectShape<IConfigOptions>>(
+    public static getObject<K extends keyof IConfigOptions>(
         key: K,
         altCaseName?: string,
-    ): Optional<SnakedObject<IConfigOptions[K]>> {
-        const val = SdkConfig.get(key, altCaseName);
+    ): Optional<SnakedObject<NonNullable<IConfigOptions[K]>>> {
+        const val = SdkConfig.get(key, altCaseName)!;
         if (val !== null && val !== undefined) {
```

Comment:
If we are asserting that `val` is always defined (ie `!`after `SdkConfig.get(key, altCaseName)`), why do we still check that `val` is not `null` or `undefined` ?

Reply:
Spurious, good spot